funsor/distribution.py

```diff
@@ -249,23 +249,30 @@ def _infer_param_domain(cls, name, raw_shape):
         # `infer_param_domain` methods.
         # Because NumPyro and Pyro have the same pattern, we use name check for simplicity.
         support_name = type(support).__name__
+
+        event_dim = 0
+        while support_name == "_IndependentConstraint":
```

Comment:
Can you change this to 
```py
support_name = type(support).__name__.lstrip("_")

event_dim = 0
while support_name == "IndependentConstraint":
    ...
```
so as to be compatible with all backends? Note in Pyro the class is named `IndependentConstraint`.

Reply:
We could also duplicate `_infer_param_domain` in each backend so that dispatch happens on constraint types rather than names.

Reply:
> and in a future PyTorch the class will be renamed to IndependentConstraint

Does it apply to other constraints as well?

Reply:
yes, see plans in https://github.com/pytorch/pytorch/issues/50616